monai/metrics/panoptic_quality.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Union
+
+import torch
+
+from monai.metrics.metric import CumulativeIterationMetric
+from monai.metrics.utils import do_metric_reduction, remap_instance_id
+from monai.utils import MetricReduction, optional_import
+
+linear_sum_assignment, _ = optional_import("scipy.optimize", name="linear_sum_assignment")
+
+__all__ = ["PanopticQualityMetric", "compute_panoptic_quality"]
+
+
+class PanopticQualityMetric(CumulativeIterationMetric):
+    """
+    Compute Panoptic Quality between two tensors. If specifying `metric_name` to "SQ" or "RQ",
+    Segmentation Quality (SQ) or Recognition Quality (RQ) will be returned instead.
+
+    Panoptic Quality is a metric used in panoptic segmentation tasks. This task unifies the typically distinct tasks
+    of semantic segmentation (assign a class label to each pixel) and
+    instance segmentation (detect and segment each object instance). Compared with semantic segmentation, panoptic
+    segmentation distinguish different instances that belong to same class.
+    Compared with instance segmentation, panoptic segmentation does not allow overlap and only one semantic label and
+    one instance id can be assigned to each pixel.
+    Please refer to the following paper for more details:
+    https://openaccess.thecvf.com/content_CVPR_2019/papers/Kirillov_Panoptic_Segmentation_CVPR_2019_paper.pdf
+
+    Different from the original panoptic quality, this class refers to the evaluation metric of CoNiC Challenge 2022,
+    a Nuclei instance segmentation and classification task. The referred implementation can be found in:
+    https://github.com/TissueImageAnalytics/CoNIC
```

Comment:
Could you please let me know what is the main difference? Thanks

Reply:
Hi @drbeh , it is mainly for the input side. CoNIC version defines the input to be a (batch, 2, H, W) format, and the 1st channel and 2nd channel represent instance map and class map respectively. For the original version, it only focus on how to calculate on (H, W) input.

Reply:
I see. Then maybe you can remove this comment (or elaborate on the difference).